File: Classes/EVEOnlineAPI/Map/EVEKills.m
//
//  EVEKills.m
//  EVEOnlineAPI
//
//  Created by Artem Shimanski on 8/20/10.
//  Copyright 2010 __MyCompanyName__. All rights reserved.
//

#import "EVEKills.h"


@implementation EVEKillsItem

+ (NSDictionary*) scheme {
	static NSDictionary* scheme = nil;
	if (!scheme)
		scheme = @{@"solarSystemID":@{@"type":@(EVEXMLSchemePropertyTypeScalar)},
				   @"shipKills":@{@"type":@(EVEXMLSchemePropertyTypeScalar)},
				   @"factionKills":@{@"type":@(EVEXMLSchemePropertyTypeScalar)},
				   @"podKills":@{@"type":@(EVEXMLSchemePropertyTypeScalar)}};
	
	return scheme;
}

@end


@implementation EVEKills

+ (NSDictionary*) scheme {
	static NSDictionary* scheme = nil;
	if (!scheme)
		scheme = @{@"solarSystems":@{@"type":@(EVEXMLSchemePropertyTypeRowset), @"class":[EVEKillsItem class]}};
	
	return scheme;
}

@end

File: EVEAPI/EVEAPI/EVEKills.swift
//
//  EVEKills.swift
//  EVEAPI
//
//  Created by Artem Shimanski on 30.11.16.
//  Copyright © 2016 Artem Shimanski. All rights reserved.
//

import UIKit

public class EVEKillsItem: EVEObject {
	public var solarSystemID: Int = 0
	public var shipKills: Int = 0
	public var factionKills: Int = 0
	public var podKills: Int = 0
	
	public required init?(dictionary:[String:Any]) {
		super.init(dictionary: dictionary)
	}
	
	public required init?(coder aDecoder: NSCoder) {
		super.init(coder: aDecoder)
	}
	
	override public func scheme() -> [String:EVESchemeElementType] {
		return [
		"solarSystemID":EVESchemeElementType.Int(elementName:nil, transformer:nil),
		"shipKills":EVESchemeElementType.Int(elementName:nil, transformer:nil),
		"factionKills":EVESchemeElementType.Int(elementName:nil, transformer:nil),
		"podKills":EVESchemeElementType.Int(elementName:nil, transformer:nil),
		]
	}
}

public class EVEKills: EVEResult {
	public var solarSystems: [EVEKillsItem] = []
	
	public required init?(dictionary:[String:Any]) {
		super.init(dictionary: dictionary)
	}
	
	public required init?(coder aDecoder: NSCoder) {
		super.init(coder: aDecoder)
	}
	
	override public func scheme() -> [String:EVESchemeElementType] {
		return [
			"solarSystems":EVESchemeElementType.Rowset(elementName: nil, type: EVEKillsItem.self, transformer: nil)
		]
	}
}

